### PR TITLE
Lensflare: Add WebGPU version.

### PR DESCRIPTION
**Pull request by aardgoose**
Related issue: #29295

Conversion of lenflare object to TSL, some differences in color handling but the basic mechanics work

### CODE REVIEW

**Comment by Mugen87** [comment]
I have tested the change locally and yeah there are some differences in the final lensflare color:

LensflareGPU:

image

Lensflare:

image

It looks like the flares are blended differently.

**Comment by Mugen87** [comment]
Um, the new version handles color spaces more correctly. Maybe we should apply the same changes to `Lensflare` as well?

File: examples/jsm/objects/LensflareGPU.js [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+	MeshBasicNodeMaterial,
+	NodeMaterial,
+	UnsignedByteType,
+	Vector2,
+	Vector3,
+	Vector4 } from 'three';
+
+import { texture, textureLoad, uv, ivec2, vec2, vec4, positionGeometry, reference, varyingProperty, materialReference, Fn, Node } from 'three/tsl';
+
+class Lensflare extends Mesh {
```
Comment (Mugen87) [inline examples/jsm/objects/LensflareGPU.js:19]: According to https://github.com/mrdoob/three.js/pull/29019#issuecomment-2259786334, we might want to rename the module to `LensflareMesh`.

File: examples/jsm/objects/LensflareGPU.js [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+			visibility.addAssign( textureLoad( occlusionMap, ivec2( 2, 8 ) ) );
+			visibility.addAssign( textureLoad( occlusionMap, ivec2( 8, 8 ) ) );
+
+			const vVisibility = varyingProperty( 'float', 'vVisibility' );
+
+			vVisibility.assign( visibility.r.div( 9.0 ) );
+			vVisibility.mulAssign( visibility.g.div( 9.0 ).oneMinus() );
+			vVisibility.mulAssign( visibility.b.div( 9.0 ) );
+
+			return vec4( ( pos.mul( scale ).add( screenPosition.xy ).xy ), 0, 1.0 );
```
Comment (Mugen87) [inline examples/jsm/objects/LensflareGPU.js:127]: Why did you change the `z` coordinate to `0`? So far the module has been using `screenPosition.z`.

Comment (aardgoose) [reply examples/jsm/objects/LensflareGPU.js:127]: junk change from debugging. i'll fix

File: examples/webgpu_lensflares.html [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+					lensflare.addElement( new LensflareElement( textureFlare3, 70, 0.7 ) );
+					lensflare.addElement( new LensflareElement( textureFlare3, 120, 0.9 ) );
+					lensflare.addElement( new LensflareElement( textureFlare3, 70, 1 ) );
+					light.add( lensflare );
+
+				}
+
+				// renderer
+
+				renderer = new THREE.WebGPURenderer( { antialias: false, forceWebGL: true } );
```
[the remark this answers is not in the document]
Comment (aardgoose) [reply examples/webgpu_lensflares.html:127]: will do

File: examples/jsm/objects/LensflareGPU.js (hunk 1 above)
Comment (aardgoose) [reply examples/jsm/objects/LensflareGPU.js:19]: sounds reasonable.

File: examples/jsm/objects/LensflareGPU.js (hunk 2 above)
Comment (aardgoose) [reply examples/jsm/objects/LensflareGPU.js]: Ive returned it to the original, but screenPosition.z is never set, so is 0 anyway.  Maybe make it a Vector2().?

Comment (Mugen87) [reply examples/jsm/objects/LensflareGPU.js]: It is set here:

https://github.com/mrdoob/three.js/blob/70ed81ad46aa52d718fdedc8bb9b129e708f20d5/examples/jsm/objects/Lensflare.js#L231

`positionScreen` is a 3D vector with a valid z value. 

This needs to be corrected in `LensflareGPU`.

**Comment by Mugen87** [comment]
Let's merge this! Color spaces are handled more correctly in `LensflareMesh` but I think it's better for backwards-compatibility to not change the old `Lensflare` implementation at the moment.
